### DISCUSSION TITLE
Add an option to automatically create keybindings for numpad keys

### DISCUSSION THREAD

**Discussion by Maykeye** [YaLTeR/niri#595] [Ideas]
On laptop I have convenient numpad  and inconvenient pageup/pagedown so I want to use numpad as well. However numpad keys and non numpad keys are not the same.

As the result right now for each numpad key I need to use three entries: the main key, numpad key(num lock off), numpad key(numlock on)

```
    Mod+KP_Prior       { focus-workspace-up; }
    Mod+KP_Next        { focus-workspace-down; }
    Mod+KP_9           { focus-workspace-up; }
    Mod+KP_3           { focus-workspace-down; }
    Mod+Page_Down      { focus-workspace-down; }
    Mod+Page_Up        { focus-workspace-up; }
````

It would be cool to have an option to generate them automatically so I don't have to bloat the config. Ideally make it so numlock can be ignored, ie have an option something like `auto-numpad-keybinding`  with  options lke

* none: current behavior 
* numlock
for every binding that uses 9, create a keybinding(if it doesn't exist) with KP_9 instead
for every binding that uses PageUp, create a keybinding(if it doesn't exist) with KP_Prior instead, 
*  digits
for every binding that uses 9, create two  keybindings(if they don't exist): one  with KP_9 instead of 9 and another with KP_Prior instead of 9.

*  navigation
for every binding that uses PageUp, create two  keybindings(if they don't exist): one  with KP_9 instead of 9 and another with KP_Prior instead of 9.

and either have it as a separate option to apply to all bindings, or make it as a property:

```
Mod+Page_Up clone-for-numpad="navigation"       { focus-workspace-up; }
Mod+Page_Down clone-for-numpad="navigation"     { focus-workspace-down; }
```

**Comment by YaLTeR** [comment]
Hmm, potentially interesting, but also a bit complex with all these choices.

Also, can these keys be hardcoded? Or do different layouts have different keys on the numpad?

**Comment by Maykeye** [comment]
Grepping around /usr/share/X11/xkb, it seems there are some weird layouts like [German's neo from 2017]( https://en.wikipedia.org/wiki/File:Kommerziell_produzierte_NEO-Tastatur_2017.JPG) and that  there are some keyboards that use Arabic_9 instead of KP_9 and that there are some Japanese layouts with numbers only(if such layouts  are actually used, for them numpad doesn't even matter)
